src/components/Canvas.tsx

```diff
@@ -29,22 +30,28 @@ function renderTopDown(node) {
   node.children.forEach(c => renderTopDown(c))
 }
 
-export const Canvas = (props) => {
+export interface CanvasOptions {
+  coreExtensionModule?: string,
+  threadXCoreWorkerUrl?: string,
+}
```

Comment:
Most of it can! I don't want to couple the ThreadXRenderDriver into the Renderer's initialization because I want applications to have the option to tree shake away all the ThreadX stuff if they don't want to use it. So the importing and initialization of the ThreadXRenderDriver need to happen at either the application itself or in Solid.